### DISCUSSION TITLE
Bug，for convertion to onnx, from pth

### DISCUSSION THREAD

**Discussion by YAOXIN77** [RapidAI/RapidLatexOCR#8] [Q&A] [answered]
Model Conversion Notes
[Convert the decoder.onnx:](https://github.com/RapidAI/RapidLatexOCR/wiki/Model-Conversion-Notes#convert-the-decoderonnx)

self.net,
(x, mask, context),
....

this line has an error: we don't have "context"!
please let me know how to solve it, thanks very much, I really need it!
Thanks a lot!

**Comment by YAOXIN77** [comment]
And then, when i try to solve it with adding "context = kwargs.get('context')" above, another assertion issue happened:
  File "/opt/conda/envs/test2/lib/python3.9/site-packages/x_transformers-0.15.0-py3.9.egg/x_transformers/x_transformers.py", line 593, in forward
    assert not (self.cross_attend ^ exists(context)), 'context must be passed in if cross_attend is set to True'
AssertionError: context must be passed in if cross_attend is set to True

I cannot do more, and please tell me how to deal with it, thank u!!!!!!

**Comment by SWHL** [comment]
Because it has been so long since the model was converted, I don't know the cause of this error.

What I can do is add the specific versionsd of the packages involved in the [conversions notes](https://github.com/RapidAI/RapidLatexOCR/wiki/Model-Conversion-Notes#my-requirements-only-some-key-packages), hoping it can help you.

**Comment by SWHL** [comment]
In the next few days, I will start to reorganize the notes related to converting this model to make it easier for everyone to reproduce. 

Please wait a little longer!

**Comment by SWHL** [comment, marked as answer]
I have reorganized the conversion code into this project [ConvertLaTeXOCRToONNX](https://github.com/SWHL/ConvertLaTeXOCRToONNX).
Enjoy it!
